protocol/communities_messenger_shared_member_address_test.go

```diff
@@ -187,6 +178,59 @@ func (s *MessengerCommunitiesSharedMemberAddressSuite) createEditSharedAddresses
 	return request
 }
 
+func (s *MessengerCommunitiesSharedMemberAddressSuite) joinOnRequestCommunityAsTokenMaster(community *communities.Community) {
+	bobRequest := createRequestToJoinCommunity(&s.Suite, community.ID(), s.bob, bobPassword, []string{bobAddress})
+	requestToJoinID := requestToJoinCommunity(&s.Suite, s.owner, s.bob, bobRequest)
+	// accept join request
+	acceptRequestToJoin := &requests.AcceptRequestToJoinCommunity{ID: requestToJoinID}
+	response, err := s.owner.AcceptRequestToJoinCommunity(acceptRequestToJoin)
+	s.Require().NoError(err)
+	s.Require().NotNil(response)
+
+	updatedCommunity := response.Communities()[0]
+	s.Require().NotNil(updatedCommunity)
+	s.Require().True(updatedCommunity.HasMember(&s.bob.identity.PublicKey))
+	s.Require().True(updatedCommunity.IsMemberTokenMaster(&s.bob.identity.PublicKey))
+
+	// receive request to join response
+	_, err = WaitOnMessengerResponse(
+		s.bob,
+		func(r *MessengerResponse) bool {
+			return len(r.Communities()) > 0 &&
+				r.Communities()[0].HasMember(&s.bob.identity.PublicKey) &&
+				r.Communities()[0].IsMemberTokenMaster(&s.bob.identity.PublicKey)
+		},
+		"user did not receive request to join response",
+	)
+	s.Require().NoError(err)
+
+	userCommunity, err := s.bob.GetCommunityByID(community.ID())
+	s.Require().NoError(err)
+	s.Require().True(userCommunity.HasMember(&s.bob.identity.PublicKey))
+	s.Require().True(userCommunity.IsTokenMaster())
+}
+
+func (s *MessengerCommunitiesSharedMemberAddressSuite) waitForRevealedAddresses(receiver *Messenger, communityID types.HexBytes, expectedAccounts []*protobuf.RevealedAccount) {
+	_, err := WaitOnMessengerResponse(receiver, func(r *MessengerResponse) bool {
+		revealedAccounts, err := receiver.communitiesManager.GetRevealedAddresses(communityID, s.alice.IdentityPublicKeyString())
+		if err != nil {
+			return false
+		}
+		if len(expectedAccounts) != len(revealedAccounts) {
+			return false
+		}
+
+		for index := range revealedAccounts {
+			if revealedAccounts[index].Address != expectedAccounts[index].Address {
```

Comment:
Are we sure to expect the same order of accounts?  Maybe better to check by maps or smth?